From 45db6476cefe3e3ba1e22a9d490c83039babc2e9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 05 Dec 2018 14:41:12 +0200
Subject: [PATCH] spi: move spi functions to spi namespace

Make more order in naming convention and move the spi related functions
to 'spi_' namespace.

Change-Id: Ib5cdc387a9e682633561531f88c438a4f202c467
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index d264160..b2da5f0 100644
--- a/preboot.c
+++ b/preboot.c
@@ -158,7 +158,7 @@
  * mmPSOC_GPIO serve 16 GPIOs
  * Note: CS0 is active low
  */
-static void hl_ssi_cs0_init(void)
+static void spi_ssi_cs0_init(void)
 {
 	unsigned int reg, mask = BIT_MASK(53 % 16);
 
@@ -171,12 +171,12 @@
 	writel(reg, mmPSOC_GPIO3_OUTENSET);
 }
 
-static void board_spi_claim_bus(void)
+static void spi_claim_bus(void)
 {
 	writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_53);
 }
 
-static void board_spi_release_bus(void)
+static void spi_release_bus(void)
 {
 	writel(0, mmPSOC_GLOBAL_CONF_PAD_SEL_53);
 }
@@ -188,7 +188,7 @@
 	writel(1, mmPSOC_SPI_SSIENR);
 }
 
-static inline int fifo_is_empty(unsigned long level_reg)
+static inline int spi_fifo_is_empty(unsigned long level_reg)
 {
 	return !readl(level_reg);
 }
@@ -197,26 +197,26 @@
 {
 	unsigned int tmp;
 
-	while (!fifo_is_empty(mmPSOC_SPI_TXFLR));
+	while (!spi_fifo_is_empty(mmPSOC_SPI_TXFLR));
 	writel(cmd, mmPSOC_SPI_DR0);
 
-	while (fifo_is_empty(mmPSOC_SPI_RXFLR));
+	while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
 	tmp = readl(mmPSOC_SPI_DR0);
 
 	if (val)
 		*val = tmp;
 }
 
-static void hl_spi_special_cmd(unsigned int cmd, unsigned char *val)
+static void spi_special_cmd(unsigned int cmd, unsigned char *val)
 {
 	unsigned char tmp;
 
 	spi_dev_update_frame_size(sizeof(tmp));
 
-	while (!fifo_is_empty(mmPSOC_SPI_TXFLR));
+	while (!spi_fifo_is_empty(mmPSOC_SPI_TXFLR));
 	writeb(cmd, mmPSOC_SPI_DR0);
 
-	while (fifo_is_empty(mmPSOC_SPI_RXFLR));
+	while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
 	tmp = readb(mmPSOC_SPI_DR0);
 
 	if (val)
@@ -224,86 +224,86 @@
 }
 
 /* Hold chip select (CS) during transaction */
-static void hl_spi_char_cmd_csel(unsigned char cmd, unsigned char *val)
+static void spi_char_cmd_csel(unsigned char cmd, unsigned char *val)
 {
-	board_spi_claim_bus();
+	spi_claim_bus();
 
-	hl_spi_special_cmd(cmd, NULL);
+	spi_special_cmd(cmd, NULL);
 	if (val)
-		hl_spi_special_cmd(0, val);
+		spi_special_cmd(0, val);
 
-	board_spi_release_bus();
+	spi_release_bus();
 }
 
-static void hl_spi_special_cmd_csel(unsigned int cmd, unsigned char *val, size_t size)
+static void spi_special_cmd_csel(unsigned int cmd, unsigned char *val, size_t size)
 {
 	int i;
 	unsigned char tmp;
 
-	board_spi_claim_bus();
+	spi_claim_bus();
 
 	spi_dev_update_frame_size(sizeof(*val));
 
-	while (!fifo_is_empty(mmPSOC_SPI_TXFLR));
+	while (!spi_fifo_is_empty(mmPSOC_SPI_TXFLR));
 	writeb(cmd, mmPSOC_SPI_DR0);
 	for (i=0; i<size; i++) {
-		while (!fifo_is_empty(mmPSOC_SPI_TXFLR));
+		while (!spi_fifo_is_empty(mmPSOC_SPI_TXFLR));
 		writeb(val[i], mmPSOC_SPI_DR0);
 	}
 
-	while (fifo_is_empty(mmPSOC_SPI_RXFLR));
+	while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
 	tmp = readb(mmPSOC_SPI_DR0);
 	for (i=0; i<size; i++) {
-		while (fifo_is_empty(mmPSOC_SPI_RXFLR));
+		while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
 		tmp = readb(mmPSOC_SPI_DR0);
 	}
 
-	board_spi_release_bus();
+	spi_release_bus();
 }
 
-static void hl_spi_common_write(unsigned int cmd, unsigned char *buf, size_t size)
+static void spi_common_write(unsigned int cmd, unsigned char *buf, size_t size)
 {
 	unsigned char sr;
 
-	hl_spi_char_cmd_csel(CMD_WRITE_ENABLE, NULL);
-	hl_spi_special_cmd_csel(cmd, buf, size);
+	spi_char_cmd_csel(CMD_WRITE_ENABLE, NULL);
+	spi_special_cmd_csel(cmd, buf, size);
 	do
-		hl_spi_char_cmd_csel(CMD_READ_STATUS, &sr);
+		spi_char_cmd_csel(CMD_READ_STATUS, &sr);
 	while (sr & READ_STATUS_WIP);
-	hl_spi_char_cmd_csel(CMD_WRITE_DISABLE, NULL);
+	spi_char_cmd_csel(CMD_WRITE_DISABLE, NULL);
 }
 
-static unsigned char hl_spi_prot(void)
+static unsigned char spi_prot(void)
 {
 	unsigned char sr, cr, data[2];
 
-	hl_spi_char_cmd_csel(CMD_READ_CONF, &cr);
-	hl_spi_char_cmd_csel(CMD_READ_STATUS, &sr);
+	spi_char_cmd_csel(CMD_READ_CONF, &cr);
+	spi_char_cmd_csel(CMD_READ_STATUS, &sr);
 	sr |= PROT_MASK;
 	data[0] = sr;
 	data[1] = cr;
-	hl_spi_common_write(CMD_WRITE_STATUS, data, sizeof(data));
+	spi_common_write(CMD_WRITE_STATUS, data, sizeof(data));
 	return sr;
 }
 
-static void hl_spi_read_spl(unsigned int offset)
+static void spi_read_spl(unsigned int offset)
 {
 	void *dst = (void*)CONFIG_SPL_TEXT_BASE;
 	unsigned int cmd = (CMD_READ << 24) | offset;
 	int i;
 
-	board_spi_claim_bus();
+	spi_claim_bus();
 	spi_dev_update_frame_size(sizeof(cmd));
 	spi_dev_dword_cmd(cmd, NULL);
 	for (i=0; i < SPL_SIZE; i += sizeof(unsigned int))
 		spi_dev_dword_cmd(0, dst + i);
 	spi_dev_dword_cmd(0, NULL);
-	board_spi_release_bus();
+	spi_release_bus();
 }
 
-static void hl_spi_init(void)
+static void spi_init(void)
 {
-	hl_ssi_cs0_init();
+	spi_ssi_cs0_init();
 	writel(0x00000000, mmPSOC_SPI_SSIENR);
 	writel(0x00000002, mmPSOC_SPI_BAUDR);	/* Min divisor */
 	writel(0x00000000, mmPSOC_SPI_RXFTLR);
@@ -320,9 +320,9 @@
 
 	preboot_copy_prepare(&spl_offset);
 
-	hl_spi_init();
-	hl_spi_read_spl(spl_offset);
-	sr = hl_spi_prot();
+	spi_init();
+	spi_read_spl(spl_offset);
+	sr = spi_prot();
 	preboot_puts("SPL copying from flash is completed\n");
 	if ((sr & PROT_MASK) == PROT_MASK)
 		preboot_puts("Flash may be protected, depend on WP# signal\n");
